test/e2e-secondary-network/framework.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	antreae2e "antrea.io/antrea/test/e2e"
+	corev1 "k8s.io/api/core/v1"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+type TestData struct {
+	e2eTestData        *antreae2e.TestData
+	logsDirForTestCase string
+}
+
+func createDirectory(path string) error {
+	return os.Mkdir(path, 0700)
+}
+
+const (
+	busyboxImage    = "projects.registry.vmware.com/antrea/busybox"
+	defaultInterval = 1 * time.Second
+)
+
+var testData *TestData
+
+type ClusterInfo struct {
+	controlPlaneNodeName string
+}
+
+var clusterInfo ClusterInfo
+
+// podWaitForRunning polls the k8s apiserver until the specified Pod is in the "running" state (or
+// until the provided timeout expires).
+func (data *TestData) podWaitForRunning(timeout time.Duration, name, namespace string) error {
```

Comment:
Question - do we really need to wrap these funcs, compared to directly calling `data.e2eTestData.Func()` in code?

Reply:
we create few wrappers with references to how a similar scenario was handled at the multicluster test implementation. removed this now and addressed as suggested. Please note: updated code was not tested (just compiled) due to K8s cluster setup issues. I will test the e2e flows once its recovered.